test/scripts/e2e_client_runner.py

```diff
@@ -79,15 +79,21 @@ def _script_thread_inner(runset, scriptname):
     # send one million Algos to the test wallet's account
     params = algod.suggested_params()
     round = params['lastRound']
-    txn = algosdk.transaction.PaymentTxn(sender=maxpubaddr, fee=params['minFee'], first=round, last=round+100, gh=params['genesishashb64'], receiver=addr, amt=1000000000000, flat_fee=True)
+    max_init_wait_rounds = 5
+    txn = algosdk.transaction.PaymentTxn(sender=maxpubaddr, fee=params['minFee'], first=round, last=round+max_init_wait_rounds, gh=params['genesishashb64'], receiver=addr, amt=1000000000000, flat_fee=True)
     stxn = kmd.sign_transaction(pubw, '', txn)
     txid = algod.send_transaction(stxn)
     ptxinfo = None
-    for i in range(50):
+    for i in range(max_init_wait_rounds):
         txinfo = algod.pending_transaction_info(txid)
         if txinfo.get('round'):
             break
-        time.sleep(0.1)
+        status = algod.status_after_block(round_num=round)
+        round = status['lastRound']
```

Comment:
I think sleep is for good here, some delay between requests is better than nothing

Reply:
The status_after_block is a blocking function, that would return when the target round+1 is reached.
I'm not sure why some delay would be better; could you expand on this one ?
(i.e. this change reduces the number of requests from ~50 to 4. sleeping won't help here )

Reply:
the status_after_block api waiting for a block is better than sleep. good change.

Reply:
did not get it is blocking call